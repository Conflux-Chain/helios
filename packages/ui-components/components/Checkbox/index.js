import PropTypes from 'prop-types'
import {Checked, NoChecked} from '../../assets/svg'

function Checkbox({checked, onChange, children, className}) {
  return (
    <div
      onClick={() => onChange && onChange()}
      aria-hidden="true"
      className={`flex items-center cursor-pointer ${className}`}
    >
      {checked && <Checked className="w-4 h-4 mr-2" />}
      {!checked && <NoChecked className="w-4 h-4 mr-2 text-gray-40" />}
      {children}
    </div>
  )
}

export default Checkbox

Checkbox.propTypes = {
  checked: PropTypes.bool,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  className: PropTypes.string,
  onChange: PropTypes.func,
}
