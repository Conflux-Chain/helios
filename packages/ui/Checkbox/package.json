{
  "name": "@cfxjs/checkbox",
  "type": "module",
  "main": "index.js"
}
