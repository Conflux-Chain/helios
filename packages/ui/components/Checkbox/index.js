import React from 'react'
import PropTypes from 'prop-types'
import {CheckSquareFilled, SquareOutlined} from '@fluent-wallet/component-icons'

function Checkbox({
  checked,
  onChange,
  children,
  className = '',
  iconClassName = '',
  disabled = false,
}) {
  return (
    <div
      data-testid="checkbox-wrapper"
      onClick={() => !disabled && onChange && onChange()}
      aria-hidden="true"
      className={`flex items-center cursor-pointer ${className}`}
    >
      {checked && (
        <CheckSquareFilled
          className={`w-4 h-4 mr-2 ${
            disabled ? 'text-gray-40' : 'text-primary'
          } ${iconClassName}`}
          data-testid="checked-svg"
        />
      )}
      {!checked && (
        <SquareOutlined
          data-testid="non-checked-svg"
          className={`w-4 h-4 mr-2 text-gray-40 ${iconClassName}`}
        />
      )}
      {children}
    </div>
  )
}

export default Checkbox

Checkbox.propTypes = {
  checked: PropTypes.bool,
  disabled: PropTypes.bool,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  className: PropTypes.string,
  iconClassName: PropTypes.string,
  onChange: PropTypes.func,
}
