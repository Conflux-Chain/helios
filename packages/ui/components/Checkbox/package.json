{
  "private": true,
  "version": "0.0.6",
  "name": "@fluent-wallet/component-checkbox",
  "type": "module",
  "main": "index.js",
  "peerDependencies": {
    "react": "^17",
    "react-dom": "^17"
  },
  "dependencies": {
    "@fluent-wallet/component-icons": "workspace:packages/ui/icons"
  }
}
