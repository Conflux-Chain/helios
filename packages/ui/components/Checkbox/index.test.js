import {render, screen, fireEvent, cleanup} from '@testing-library/react'
import {beforeEach, describe, expect, vi, it} from 'vitest'
import Checkbox from './index.js'
beforeEach(cleanup)
describe('Checkbox', () => {
  it('test snapshot', () => {
    // eslint-disable-next-line testing-library/render-result-naming-convention
    const snapshot = render(<Checkbox checked={true}>test children</Checkbox>)
    expect(snapshot).toMatchSnapshot()
  })
  it('should render checked SVG', () => {
    render(<Checkbox checked={true}>test children</Checkbox>)
    expect(screen.getByTestId('checked-svg')).toBeInTheDocument()
  })

  it('should render  non-checked SVG', () => {
    render(<Checkbox checked={false}>test children</Checkbox>)
    expect(screen.getByTestId('non-checked-svg')).toBeInTheDocument()
  })
  it('should render incoming class name', () => {
    render(
      <Checkbox checked={false} className="test-class-name">
        test children
      </Checkbox>,
    )
    expect(screen.getByTestId('checkbox-wrapper')).toHaveClass(
      'test-class-name',
    )
  })

  it('should execute incoming onChange function', () => {
    const onChange = vi.fn()
    render(<Checkbox onChange={onChange}>test children</Checkbox>)
    fireEvent.click(screen.getByTestId('checkbox-wrapper'))
    expect(onChange).toHaveBeenCalledTimes(1)
  })
})
