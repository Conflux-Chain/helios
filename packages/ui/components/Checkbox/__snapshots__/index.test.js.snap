// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox test snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-hidden="true"
        class="flex items-center cursor-pointer undefined"
        data-testid="checkbox-wrapper"
      >
        <svg
          class="w-4 h-4 mr-2"
          data-testid="checked-svg"
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="#44D7B6"
            height="20"
            rx="2"
            width="20"
            x="2"
            y="2"
          />
          <path
            d="M15.5 9.5L11.432 15 9 13.125"
            stroke="#fff"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
        test children
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-hidden="true"
      class="flex items-center cursor-pointer undefined"
      data-testid="checkbox-wrapper"
    >
      <svg
        class="w-4 h-4 mr-2"
        data-testid="checked-svg"
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="#44D7B6"
          height="20"
          rx="2"
          width="20"
          x="2"
          y="2"
        />
        <path
          d="M15.5 9.5L11.432 15 9 13.125"
          stroke="#fff"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
      test children
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
